New Chassis Updates + Autonomous

Henyahyah opened the pull request:
I updated a couple things to work with tinkered with stuff when the chassis was expanded. I also have been working on autonomous a lot.

Before issuing a pull request, please see the contributing page.

Henyahyah committed:
New Chassis Updates + Autonomous

I updated a couple things to work with tinkered with stuff when the chassis was expanded. I also have been working on autonomous a lot.

kjrokos committed:
Merge branch 'master' into NewChassisUpdates+Auton

kjrokos commented on TeamCode/src/main/java/org/firstinspires/ftc/teamcode/AutoRobotDrive.java:
```diff
@@ -171,67 +174,84 @@ void strafeRight(int ticks) {
+            forward(537);
+            pause(1);
+            viperArm.highBucket();
+            runtime.reset();
```
I don't think we need `runtime.reset()` here or on line 236. We do it in the `pause` function so we can use the timer, but we aren't using the timer in the while loops here.

kjrokos commented on TeamCode/src/main/java/org/firstinspires/ftc/teamcode/AutoRobotDrive.java:
```diff
@@ -144,23 +145,25 @@ void pause(double seconds) {
+            leftBackDrive.setPower(-.5);
+            leftFrontDrive.setPower(.5);
+            rightBackDrive.setPower(.5);
+            rightFrontDrive.setPower(-.5);
```
We have a constant called `FORWARD_SPEED` defined earlier in this file. I think we can use that for all these functions (and the diagonal code below), and it will be easier to adjust

kjrokos commented on TeamCode/src/main/java/org/firstinspires/ftc/teamcode/ViperArrrrrm.java:
```diff
@@ -30,11 +30,17 @@ void home(){
+    boolean isMoving() {
+        int tolerance = 50;
+        return viperArm.getCurrentPosition() > viperArm.getTargetPosition() - tolerance &&
+                viperArm.getCurrentPosition() < viperArm.getTargetPosition() + tolerance;
```
We (I) wrote this backwards. The function is called `isMoving`, but we wrote it so that it is true when the current position is near the target position. So that was why it wasn't waiting when we last tested it.

Henyahyah committed:
AUotnmus wokrs

I got autonomous to work now, just gotta test it on the field to tune it right.
